test_conformance/spirv_new/spirv_asm/linkage_linkonce_odr_noa_main.spvasm32

```diff
@@ -0,0 +1,40 @@
+; SPIR-V
+; Version: 1.0
+; Generator: Khronos LLVM/SPIR-V Translator; 14
+; Bound: 27
+; Schema: 0
+               OpCapability Addresses
+               OpCapability Linkage
+               OpCapability Kernel
+               OpExtension "SPV_KHR_linkonce_odr"
+               OpMemoryModel Physical32 OpenCL
+               OpEntryPoint Kernel %17 "test_linkonce_odr" %__spirv_BuiltInGlobalInvocationId
+               OpDecorate %__spirv_BuiltInGlobalInvocationId BuiltIn GlobalInvocationId
+               OpDecorate %__spirv_BuiltInGlobalInvocationId Constant
+               OpDecorate %18 FuncParamAttr NoCapture
+               OpDecorate %b LinkageAttributes "b" Import

```

Comment:
I think you want to use `a` which is exported using `LinkOnceODR` here and not `b` that is exported using `Export`.

Reply:
Good catch, previously Clang and the SPIR-V LLVM Translator were being too smart and removing the call to `a` completely, which meant the test wasn't actually doing what I was intending for it to be doing.

I've added back the calls to function `a` now in both cases, and added the right `Import` linkage attribute in the `noa` test.  I _think_ this is right now?